questionpy_sdk/webserver/state_storage.py

```diff
@@ -8,6 +8,24 @@
 
 
 def _unflatten(flat_form_data: dict[str, str]) -> dict[str, Any]:
+    """Splits the keys of a dictionary to form a new nested dictionary.
+
+    Each key of the input dictionary is a reference string of a FormElements from the Options Form.
+    These strings are split to create a nested dictionary, where each key is one part of the reference.
+    Additionally: Dictionaries with only numerical keys (Repetition Elements) are replaced by lists.
+
+    Example:
+        {'general[my_hidden]': 'foo',\n
+        'general[my_repetition][1][role]': 'OPT_1',\n
+        'general[my_repetition][1][name][first_name]': 'John '}
+
+        becomes:\n
+        {'general': {
+            'my_hidden': 'foo',\n
+            'my_repetition': [{'name': {'first_name': 'John '}, 'role': 'OPT_1'}]
+            }
+        }
```

Comment:
Wozu die expliziten `\n`? Mein IDEA zeigt es ohne ähnlich an (also genauso schlecht :frowning_face:, aber das liegt wohl an IDEA). Angeblich gibt es auch Code Blöcke: <https://stackoverflow.com/a/44973344> Scheint IDEA nicht zu können, vermutlich trotzdem sinnvoll.

Reply:
Mit der folgenden Änderung werden bei mir in PyCharm die Code Blöcke korrekt gerendert:
```suggestion
    Example:
        This::

            {
                'general[my_hidden]': 'foo',
                'general[my_repetition][1][role]': 'OPT_1',
                'general[my_repetition][1][name][first_name]': 'John '
            }
        becomes::

            {
                'my_hidden': 'foo',
                'my_repetition': [{'name': {'first_name': 'John '}, 'role': 'OPT_1'}],
            }
```

Reply:
@alexanderschmitz Der Docstring [weiter unten](https://github.com/questionpy-org/questionpy-sdk/pull/57/files#diff-8c67737775ea5d412c925fb8ef3b59efd97795ae692bce31d3ff8f71642ac002R64-R78) kann auch noch angepasst werden.